pkgs/applications/editors/neovim/default.nix

```diff
@@ -15,9 +16,11 @@ let
   neovimLuaEnv = lua.withPackages(ps:
     (with ps; [ lpeg luabitop mpack ]
     ++ optionals doCheck [
-        nvim-client luv coxpcall busted luafilesystem penlight inspect
+        nvim-client luv coxpcall busted luafilesystem penlight inspect luacheck
```

Comment:
I don't like this unneeded package added to the environment. `neovimLuaEnv` is in `buildInputs` - meaning they all fill up the closure. It would have been awesome if it could have been put into `nativeBuildInputs` or `checkInputs` somehow, along with `busted`, `penlight` and `inspect` as well.

Reply:
I've a better version of this locally (with luacheck in checkInputs as you mentioned) but failed to push because changes are mixed up with some changes necessary to compile neovim's master.
I will update this commit after 0.5 release I think. thanks.

Reply:
I've updated a clean version since 0.5 is still a long way.